opencog/haskell/OpenCog/AtomSpace/Api.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+module OpenCog.AtomSpace.Api (
+      asAddNode
+    , runOnNewAtomSpace
+    , AtomSpace
+    ) where
+
+-- Note that I don't export the AtomSpace data constructors nor the asDelete/asNew functions.
+
+import Foreign
+import Foreign.C.Types
+import Foreign.C.String
+import Control.Exception
+import OpenCog.AtomSpace.Types
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Reader
+import Data.Functor
+
+-- Internal AtomSpace reference to a mutable C++ instance of the AtomSpace class.
+newtype AtomSpaceRef = AtomSpaceRef (Ptr AtomSpaceRef)
+
+-- Main Data Type for representing programs working on an AtomSpace.
+-- We have to use the IO monad because of the use of FFI for calling c functions for working
```

Comment:
@MarcosPividori we have a coding standard defined on the wiki
http://wiki.opencog.org/w/Development_standards
try you're best to follow it, when it makes sense, for instance here the comment line is more than 80-col long.